example/oven_fsm.dart

```diff
@@ -0,0 +1,228 @@
+            led < LEDLight.blue().value,
+            counterReset < 1,
+            en < 0,
+          ]),
+
+      // identifier: cooking state, represent by `OvenState.cooking`.
+      State<OvenState>(OvenState.cooking,
+          // events:
+          // When the button `paused` is pressed during cooking state,
+          // OvenState will changed to `OvenState.paused` state.
+          //
+          // When the button `counter` time is elapsed during cooking state,
+          // OvenState will changed to `OvenState.completed` state.
+          events: {
+            Logic(name: 'button_pause')..gets(button.eq(Button.pause())):
+                OvenState.paused,
+            Logic(name: 'counter_time_complete')..gets(counter.val.eq(4)):
+                OvenState.completed
+          },
+          // actions:
+          // During the cooking state, `led` is change to yellow; timer's
+          // `counterReset` is set to 0 (Do not reset);
+          // timer's `en` is set to 1 (Enable value update).
+          actions: [
+            led < LEDLight.yellow().value,
+            counterReset < 0,
+            en < 1,
+          ]),
+
+      // identifier: paused state, represent by `OvenState.paused`.
+      State<OvenState>(OvenState.paused,
+          // events:
+          // When the button `resume` is pressed during paused state,
+          // OvenState will changed to `OvenState.cooking` state.
+          events: {
+            Logic(name: 'button_resume')..gets(button.eq(Button.resume())):
+                OvenState.cooking
+          },
+          // actions:
+          // During the paused state, `led` is change to red; timer's
+          // `counterReset` is set to 0 (Do not reset);
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.red().value,
+            counterReset < 0,
+            en < 0,
+          ]),
+
+      // identifier: completed state, represent by `OvenState.completed`.
+      State<OvenState>(OvenState.completed,
+          // events:
+          // When the button `start` is pressed during completed state,
+          // OvenState will changed to `OvenState.cooking` state.
+          events: {
+            Logic(name: 'button_start')..gets(button.eq(Button.start())):
+                OvenState.cooking
+          },
+          // actions:
+          // During the start state, `led` is change to green; timer's
+          // `counterReset` is set to 1 (Reset value);
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.green().value,
+            counterReset < 1,
+            en < 0,
+          ])
+    ];
+
+    // Assign the oven StateMachine object to public variable declared.
+    oven = StateMachine<OvenState>(clk, reset, OvenState.standby, states);
+  }
+
+  // An ovenStateMachine that represent in getter.
+  StateMachine<OvenState> get ovenStateMachine => oven;
+}
+
+Future<void> main() async {
```

Comment:
If this example is added to `test/example_test.dart`, then the `noPrint` parameter must be added like the other examples.